tests/translator/test_translator.py

```diff
@@ -321,6 +316,74 @@ def test_transform_success(self, testcase, partition_with_region):
         assert deep_sort_lists(output_fragment) == deep_sort_lists(expected)
         assert len(matches) == 0
 
+    @parameterized.expand(
+      itertools.product([
+        'explicit_api_openapi_3',
+        'api_with_auth_all_maximum_openapi_3',
+        'api_with_cors_openapi_3',
+        'api_with_gateway_responses_all_openapi_3',
+        'api_with_open_api_version',
+        'api_with_open_api_version_2',
+      ],
+      [
+       ("aws", "ap-southeast-1"),
+       ("aws-cn", "cn-north-1"),
+       ("aws-us-gov", "us-gov-west-1")
+      ] # Run all the above tests against each of the list of partitions to test against
+      )
+    )
+    @patch('samtranslator.plugins.application.serverless_app_plugin.ServerlessAppPlugin._sar_service_call', mock_sar_service_call)
+    @patch('botocore.client.ClientEndpointBridge._check_default_region', mock_get_region)
+    def test_transform_success_openapi3(self, testcase, partition_with_region):
```

Comment:
I like this idea. The other method was getting too long. I wonder if we can further improve this... but this looks good to me!